csharp/src/Ice/OutputStream.cs

```diff
@@ -1353,7 +1314,7 @@ public void WriteSpan(ReadOnlySpan<byte> span)
         /// ensures there is no gaps, the size and capacity of the stream are adjusted to the
         /// sum of all segment counts, and the tail is advanced to the end of the new segment.</summary>
         /// <param name="payload">The array segment payload to write into the stream.</param>
-        protected void WritePayload(ArraySegment<byte> payload)
+        internal void WritePayload(ArraySegment<byte> payload)
```

Comment:
We could consider renaming this method to "AddArraySegment" since that's what it does and given that unlike WriteXxx methods it doesn't copy the given data.

Reply:
Fine with me, @bernardnormier  do you have any thoughts on this?

Reply:
It doesn't matter as the next refactoring of OutgoingFrame/OutputStream will remove this method entirely.